utils/response-rejection/src/index.ts

```diff
@@ -1,63 +1,162 @@
   id: string
   values: { field: string; message: string }[]
 }
 
 export async function responseRejectionHandler(
-  responseRejection: ResponseRejection
-): Promise<number> {
+  responseRejection: RejectionResponse
+): Promise<Flatfile.JobCompleteDetails> {
   let totalRejectedRecords = 0
 
-  const results = await Promise.all(
-    responseRejection.sheets.map((sheet) => updateSheet(sheet))
-  )
-  totalRejectedRecords = results.reduce((acc, val) => acc + val, 0)
+  for (const sheet of responseRejection.sheets || []) {
+    const count = await updateSheet(sheet, responseRejection.deleteSubmitted)
+    totalRejectedRecords += count
+  }
 
-  return totalRejectedRecords
+  const message = responseRejection.message ?? getMessage(totalRejectedRecords)
+  let next
+  if (responseRejection.deleteSubmitted && totalRejectedRecords > 0) {
+    next = getNext(totalRejectedRecords, responseRejection.sheets[0].sheetId)
+  }
+
+  return {
+    outcome: {
+      buttonText: 'Close',
+      heading: totalRejectedRecords > 0 ? 'Rejected Records' : 'Success!',
+      acknowledge: true,
+      ...(next && !responseRejection.deleteSubmitted && { next }),
+      message,
+    },
+  }
 }
 
-async function updateSheet(sheet: ResponseRejectionSheet): Promise<number> {
-  if (!sheet.rejectedRecords?.length) {
-    return 0
+function getMessage(totalRejectedRecords) {
+  return totalRejectedRecords > 0
+    ? `During the data submission process, ${totalRejectedRecords} records were rejected. Please review and correct these records before resubmitting.`
+    : 'The data has been successfully submitted without any rejections. This task is now complete.'
+}
+
+function getNext(
+  totalRejectedRecords: number,
+  sheetId: string
+): Flatfile.JobOutcomeNext | undefined {
+  return totalRejectedRecords > 0
+    ? {
+        type: 'id',
+        id: sheetId,
+        query: 'searchField=submissionStatus&searchValue=rejected',
+      }
+    : undefined
+}
+
+async function updateSheet(
+  sheetRejections: SheetRejections,
+  deleteSubmitted: boolean
+): Promise<number> {
+  if (!deleteSubmitted) {
+    await addSubmissionStatusField(sheetRejections.sheetId)
   }
 
-  const rejectedRecordsIds = sheet.rejectedRecords.map((record) => record.id)
-  const sheetRecords = await api.records.get(sheet.id)
+  await processRecords(sheetRejections.sheetId, async (records) => {
+    await updateRecords(sheetRejections, records, deleteSubmitted)
+  })
 
-  const rejectedSheetRecords: Flatfile.Record_[] =
-    sheetRecords.data.records?.filter((record: Flatfile.Record_) =>
-      rejectedRecordsIds.includes(record.id)
-    )
+  return sheetRejections.rejectedRecords.length
+}
+
+async function addSubmissionStatusField(sheetId: string): Promise<void> {
+  try {
+    const { data: sheet } = await api.sheets.get(sheetId)
+    if (
+      !sheet.config.fields.some((field) => field.key === 'submissionStatus')
+    ) {
+      await api.sheets.addField(sheet.id, {
```

Comment:
Should we remove this field from the sheet when submitting? I'm sure they could strip it out if they wanted but could be nice if it was cleaned up as an option